Use default linker instead of lld when building with GCC.
Otherwise we get issue with LTO.
Author: Jarne Renders (Vrije Universiteit Brussel)
--- a/cmake/MujocoLinkOptions.cmake	2026-07-28 12:09:30.521517473 +0200
+++ b/cmake/MujocoLinkOptions.cmake	2026-07-28 12:09:38.377409396 +0200
@@ -34,18 +34,6 @@
         )
       endif()
     else()
-      set(CMAKE_REQUIRED_FLAGS "-fuse-ld=lld")
-      check_c_source_compiles("int main() {}" SUPPORTS_LLD)
-      if(SUPPORTS_LLD)
-        set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -fuse-ld=lld)
-      else()
-        set(CMAKE_REQUIRED_FLAGS "-fuse-ld=gold")
-        check_c_source_compiles("int main() {}" SUPPORTS_GOLD)
-        if(SUPPORTS_GOLD)
-          set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -fuse-ld=gold)
-        endif()
-      endif()
-
       set(CMAKE_REQUIRED_FLAGS ${EXTRA_LINK_OPTIONS} "-Wl,--gc-sections")
       check_c_source_compiles("int main() {}" SUPPORTS_GC_SECTIONS)
       if(SUPPORTS_GC_SECTIONS)
--- a/sample/cmake/MujocoLinkOptions.cmake	2026-07-28 12:09:30.525517417 +0200
+++ b/sample/cmake/MujocoLinkOptions.cmake	2026-07-28 12:09:38.419408819 +0200
@@ -34,18 +34,6 @@
         )
       endif()
     else()
-      set(CMAKE_REQUIRED_FLAGS "-fuse-ld=lld")
-      check_c_source_compiles("int main() {}" SUPPORTS_LLD)
-      if(SUPPORTS_LLD)
-        set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -fuse-ld=lld)
-      else()
-        set(CMAKE_REQUIRED_FLAGS "-fuse-ld=gold")
-        check_c_source_compiles("int main() {}" SUPPORTS_GOLD)
-        if(SUPPORTS_GOLD)
-          set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -fuse-ld=gold)
-        endif()
-      endif()
-
       set(CMAKE_REQUIRED_FLAGS ${EXTRA_LINK_OPTIONS} "-Wl,--gc-sections")
       check_c_source_compiles("int main() {}" SUPPORTS_GC_SECTIONS)
       if(SUPPORTS_GC_SECTIONS)
--- a/simulate/cmake/MujocoLinkOptions.cmake	2026-07-28 12:09:30.530517349 +0200
+++ b/simulate/cmake/MujocoLinkOptions.cmake	2026-07-28 12:09:38.455408323 +0200
@@ -34,18 +34,6 @@
         )
       endif()
     else()
-      set(CMAKE_REQUIRED_FLAGS "-fuse-ld=lld")
-      check_c_source_compiles("int main() {}" SUPPORTS_LLD)
-      if(SUPPORTS_LLD)
-        set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -fuse-ld=lld)
-      else()
-        set(CMAKE_REQUIRED_FLAGS "-fuse-ld=gold")
-        check_c_source_compiles("int main() {}" SUPPORTS_GOLD)
-        if(SUPPORTS_GOLD)
-          set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -fuse-ld=gold)
-        endif()
-      endif()
-
       set(CMAKE_REQUIRED_FLAGS ${EXTRA_LINK_OPTIONS} "-Wl,--gc-sections")
       check_c_source_compiles("int main() {}" SUPPORTS_GC_SECTIONS)
       if(SUPPORTS_GC_SECTIONS)
